os/kernel/sched/sched_processtimer.c

```diff
@@ -258,6 +290,11 @@ void sched_process_timer(void)
 	{
 		clock_timer();
 	}
+
+#if defined(CONFIG_IDLE_PANIC_TIMER)
```

Comment:
Since this feature is to be used only for kernel debugging, I think we need to check for CONFIG_DEBUG flag here. 
In the real product, this feature must get disabled. The product team has its own feature to detect system hang and restart the device after 3 mins.